csharp/ql/src/semmle/code/csharp/dataflow/internal/DataFlowImplCommon.qll

```diff
@@ -746,9 +746,22 @@ class ReturnPosition extends TReturnPosition0 {
   string toString() { result = "[" + kind + "] " + c }
 }
 
+/**
+ * Gets the enclosing callable of `n`. Unlike `n.getEnclosingCallable()`, this
+ * predicate ensures that joins go from `n` to the result instead of the other
+ * way around.
+ */
+pragma[inline]
+DataFlowCallable getNodeEnclosingCallable(Node n) {
+  exists(Node n0 |
+    pragma[only_bind_into](n0) = n and
+    result = n0.getEnclosingCallable()
+  )
```

Comment:
This may work, but I think it possibly makes more sense to disconnect the callable rather than the node. I.e. like this:
```suggestion
  pragma[only_bind_into](result) = n.getEnclosingCallable()
```
I suspect this will also work, and I believe it's safer.